docs/ADR-48-locking-collections-in-the-builder.md

```diff
@@ -0,0 +1,63 @@
+
+Publications vary for each type described above:
+
+- Decentraland: You publish the entire collection with all it's items
+- Third Party: You publish each item individually, but you can do it in batches
+
+## Problem
+
+Publishing is done by sending a meta-transaction. This means that the user will send a transaction and send it to our server where it will ultimately be sent to the corresponding Polygon network to be processed.
+This process is usually pretty quick, but it can still take a while. It's a combination of a remote request to our server, and waiting for a transaction to be confirmed. The problem is therefore, that while we're waiting for the transaction to complete we should **lock** the collection/items that are being published, to avoid having missmatching data between the blockchain and our database.
+
+### Solution
+
+We propose a locking mechanism for both types of Collections and Items. We'll lock the collection/items for a day while the publication is still going. The lock is released after that if the publication ends successfully. The idea behind it is to lock the UI while the publish transaction is loading, preventing the user from making changes and the data to be in a missmatched state.
+
+To do this we'll have to put two mechanisms in place, for both Decentraland and Third Party data:
+
+- **Decentraland**: Once the user confirms the signature to send the publication, we'll send a POST to the server locking the collection (`/collections/COLLECTION_ID/lock`). This will set the current timestamp (`Date.now()`) on `lock: timestamp` property of the Collection. This will be then checked in the server when returning each Decentraland Collection, adding a `isLocked: boolean` property to each object if the publication is still pending and a day has not gone by:
+
+```ts
+function isLocked(collection: Collection) {
+  if (collection.isPublished) {
+    return false;
+  }
+
+  const deadline = new Date(lock);
+  deadline.setDate(deadline.getDate() + 1);
+
+  return deadline.getTime() > Date.now();
+}
+```
+
+- **Third Party**: Third party collections work in a similar way but a key difference outlined before, we publish items not collections. Still the process works the same way. After the user signs the transaction that will lock the items, we send a POST to the server (`/collections/COLLECTION_ID/lock`) locking each the Thrid Party Collection. This will set the current timestamp (`Date.now()`) on `lock: timestamp` property of each the Collection. Up until this point, things are mostly the same, but when the server returns the data it will now check the lock as before and if the lock is smaller than the last item created at date on the graph, which is the same as saying "last publish date":
+
+```ts
+function isLocked(collection: Collection, lastItem: Item) {
+  const deadline = new Date(lock)
+  deadline.setDate(deadline.getDate() + 1)
+
+  return deadline.getTime() > Date.now()  || collection.lock <= lastItem.created_at)
```

Comment:
Should we clarify how are we going to get the last item date? Like, we're going to query the graph for the last time an item was published for that collection.